data_validation/query_builder/query_builder.py

```diff
@@ -486,8 +486,16 @@ def compile(self, data_client, schema_name, table_name):
                 calc_table = calc_table.mutate(
                     self.compile_calculated_fields(calc_table, n)
                 )
-        if self.comparison_fields:
-            calc_table = calc_table.mutate(self.compile_comparison_fields(calc_table))
+
+        if validation_type == consts.ROW_VALIDATION:
+            calc_table = calc_table.projection(
```

Comment:
Does there need to be an `if self.comparison_fields` conditional here? Or will row validations always have a comparison field populated here?

Reply:
After testing, it seems it will always have a comparison_field